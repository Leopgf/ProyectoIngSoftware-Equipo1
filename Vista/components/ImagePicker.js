import React, { useState, useEffect } from 'react';
import { Image, View, Platform, Dimensions } from 'react-native';
import * as ImagePicker from 'expo-image-picker';
import { StyleSheet } from 'react-native';
import { Button } from 'galio-framework';
import { cloudinary } from 'cloudinary-react';

export default function ImagePickerExample({ onImagePicked, defaultImage }) {
  const { width, height } = Dimensions.get('screen');
  const [image, setImage] = useState(null);

  useEffect(() => {
    (async () => {
      if (Platform.OS !== 'web') {
        const { status } = await ImagePicker.requestCameraRollPermissionsAsync();
        if (status !== 'granted') {
          alert('Sorry, we need camera roll permissions to make this work!');
        }
      }
    })();
  }, []);

  const pickImage = async () => {
    let result = await ImagePicker.launchImageLibraryAsync({
      mediaTypes: ImagePicker.MediaTypeOptions.All,
      allowsEditing: true,
      aspect: [4, 3],
      quality: 1,
    });

    if (!result.cancelled) {
      let file = {
        uri: result.uri,
        type: `test/${result.uri.split('.').pop()}`,
        name: result.uri,
      };

      let apiUrl = 'https://api.cloudinary.com/v1_1/mixosteam/image/upload';

      const formData = new FormData();
      formData.append('file', file);
      formData.append('upload_preset', 'zc9ttas7');
      formData.append('cloud_name', 'mixosteam');

      await fetch(apiUrl, {
        body: formData,
        method: 'POST',
      })
        .then(async (r) => await r.json())
        .then((dato) => {
          console.log('TOY AQUIIII');
          setImage(dato.secure_url);
          onImagePicked({ uri: dato.secure_url });
        })
        .catch((error) => {
          console.log(error);
        });
    }
  };

  const styles = StyleSheet.create({
    smallButton: {
      width: 75,
      height: 28,
    },
    Button: {
      width: width / 2.5,
      height: height / 20,
      shadowOpacity: 0,
    },
  });

  return (
    <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center',shadowOpacity: 0}}>
      <Button style={styles.Button} onPress={pickImage}>
        Subir imagen
      </Button>
      {image && (
        <Image source={{ uri: image }} style={{ width: 200, height: 200, marginTop: 20 }} />
      )}
      {defaultImage && !image && (
        <Image source={{ uri: defaultImage }} style={{ width: 200, height: 200, marginTop: 20 }} />
      )}
    </View>
  );
}
