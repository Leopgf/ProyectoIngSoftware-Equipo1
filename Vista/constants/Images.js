// local imgs
const Onboarding = require('../assets/imgs/FFondo.jpg');
const Logo = require('../assets/imgs/logop.png');
const ProfileBackground = require('../assets/imgs/basePerfil.jpg');
const RegisterBackground = require('../assets/imgs/fondoInicio.jpg');
const Pro = require('../assets/imgs/bg.png');
const NowLogo = require('../assets/imgs/logo.png');
const iOSLogo = require('../assets/imgs/ios.png');
const androidLogo = require('../assets/imgs/android.png');
const ProfilePicture = require('../assets/imgs/profile-img.jpg');
const Conversor = require('../assets/imgs/conversor.jpg');
const AddReceta = require('../assets/imgs/addReceta.jpg');
const CreativeTimLogo = require('../assets/imgs/creative-tim-white-slim2.png');
const InvisionLogo = require('../assets/imgs/invision-white-slim.png');

const Viewed = [
  require('../assets/imgs/project5.jpg'),
  require('../assets/imgs/project7.jpg'),
  require('../assets/imgs/project6.jpg'),
  require('../assets/imgs/project4.jpg'),
  require('../assets/imgs/project21.jpg'),
  require('../assets/imgs/project24.jpg')
];

const Products = {
  text: 'View article',
  path: require('../assets/imgs/bg15.jpg')
};

export default {
  Onboarding,
  Logo,
  ProfileBackground,
  Conversor,
  ProfilePicture,
  RegisterBackground,
  AddReceta,
  Viewed,
  Pro,
  Products,
  NowLogo: NowLogo
};
