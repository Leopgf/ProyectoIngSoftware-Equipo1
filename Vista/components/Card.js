//IMPORT
import React from 'react';
import { withNavigation } from '@react-navigation/compat';
import PropTypes from 'prop-types';
import { Button as GaButton } from 'galio-framework';
import { StyleSheet, Image, TouchableWithoutFeedback } from 'react-native';
import { Block, Text, theme } from 'galio-framework';
import { nowTheme } from '../constants';

import * as firebase from 'firebase';

//CLASE DE CARD
class Card extends React.Component {
   render() {
    const {
      navigation,
      item,
      horizontal,
      full,
      style,
      ctaColor,
      imageStyle,
      ctaRight,
      titleStyle,
      params,
    } = this.props;

    const imageStyles = [full ? styles.fullImage : styles.horizontalImage, imageStyle];
    const titleStyles = [styles.cardTitle, titleStyle];
    const cardContainer = [styles.card, styles.shadow, style];
    const imgContainer = [
      styles.imageContainer,
      horizontal ? styles.horizontalStyles : styles.verticalStyles,
      styles.shadow,
    ];

    var user = '';

    if(item.userID){
      firebase.firestore().collection('Usuarios').where('usuarioID', '==', item.userID).get().then((usuarios) => {

        usuarios.forEach((usuario) => {
          user = usuario.data().usuario;
        });
      });
    }

    if(item.nombre){
      return (
        <Block row={horizontal} card flex style={cardContainer}>
          <TouchableWithoutFeedback onPress={() => navigation.navigate('DetallesReceta', params)}>
            <Block flex style={imgContainer}>
              <Image
                resizeMode="cover"
                source={item.imagen && { uri: item.imagen }}
                style={imageStyles}
              />
            </Block>
          </TouchableWithoutFeedback>
  
          <TouchableWithoutFeedback onPress={() => navigation.navigate('DetallesReceta', params)}>
            <Block flex space="between" style={styles.cardDescription}>
              <Block flex>
                <Block flex style={{ justifyContent: 'center' }}>
                  {item.nombre && !horizontal ? (
                    <Text size={24} style={styles.cardTitle} color={'#e63746'}>
                      {/*TITULO DE LA RECETA*/}
                      {item.nombre}
                    </Text>
                  ) : (
                    <Text size={24} style={styles.cardTitle2} color={'#e63746'}>
                      {/*TITULO DE LA RECETA*/}
                      {item.nombre}
                    </Text>
                  )}
                </Block>
                {item.subtitle ? (
                  <Block flex center>
                    <Text
                      style={{ fontFamily: 'montserrat-regular' }}
                      size={32}
                      color={nowTheme.COLORS.BLACK}
                    >
                      {item.subtitle}
                    </Text>
                  </Block>
                ) : null}
                {/*DESCRIPCION RECETA*/}
                {item.descripcion && !horizontal ? (
                  <Block flex center>
                    <Text
                      style={{ fontFamily: 'montserrat-regular', textAlign: 'left', padding: 15 }}
                      size={14}
                      color={'#0f1e2e'}
                    >
                      {item.descripcion}
                    </Text>
                  </Block>
                ) : null}
                {item.body ? (
                  <Block flex left>
                    <Text
                      style={{ fontFamily: 'montserrat-regular' }}
                      size={12}
                      color={nowTheme.COLORS.TEXT}
                    >
                      {item.body}
                    </Text>
                  </Block>
                ) : null}
              </Block>
              {ctaRight ? (
                <Block right={ctaRight ? true : false}>
                  <Text
                    style={styles.articleButton}
                    size={12}
                    muted={!ctaColor}
                    color={ctaColor || nowTheme.COLORS.ACTIVE}
                    bold
                  >
                    {item.cta}
                  </Text>
                </Block>
              ) : null}
            </Block>
          </TouchableWithoutFeedback>
        </Block>
      );
    }else{
        return (
          <Block row={horizontal} card flex style={cardContainer}>
            <Block flex style={imgContainer}>
              <Image
                resizeMode="cover"
                source={item.imagen && { uri: item.imagen }}
                style={imageStyles}
              />
            </Block>
  
            <Block flex space="between" style={styles.cardDescription}>
              <Block flex>
                <Block flex style={{ justifyContent: 'center' }}>
                  {item.titulo && !horizontal ? (
                    <Text size={24} style={styles.cardTitle} color={'#e63746'}>
                      {/*TITULO DE LA REVIEW*/}
                      {item.titulo}
                    </Text>
                  ) : (
                    <Text size={24} style={styles.cardTitle2} color={'#e63746'}>
                      {/*TITULO DE LA REVIEW*/}
                      {item.titulo}
                    </Text>
                  )}
                </Block>
                {/*DESCRIPCION REVIEW*/}
                {item.mensaje ? (
                  <Block flex center>
                    <Text
                      style={{ fontFamily: 'montserrat-regular', textAlign: 'left', padding: 15 }}
                      size={14}
                      color={'#0f1e2e'}
                    >
                      {item.mensaje}
                    </Text>
                  </Block>
                ) : null}
                {item.userID && item.valoracion ? (
                  <Block flex left>
                    <Text
                      style={{ fontFamily: 'montserrat-regular' }}
                      size={12}
                      color={nowTheme.COLORS.TEXT}
                    >
                      {item.userID}
                    </Text>
                    <Text
                      style={{ fontFamily: 'montserrat-regular' }}
                      size={12}
                      color={nowTheme.COLORS.TEXT}
                    >
                      {item.valoracion}
                    </Text>
                    <GaButton
                          round
                          onlyIcon
                          shadowless
                          icon="star"
                          iconFamily="Font-Awesome"
                          iconColor={'#E63746'}
                          iconSize={nowTheme.SIZES.BASE * 1.4}
                          color={'#FFFFFF'}
                          style={[styles.social, styles.shadow]}
                        />
                  </Block>
                ) : null}
              </Block>
              {ctaRight ? (
                <Block right={ctaRight ? true : false}>
                  <Text
                    style={styles.articleButton}
                    size={12}
                    muted={!ctaColor}
                    color={ctaColor || nowTheme.COLORS.ACTIVE}
                    bold
                  >
                    {item.cta}
                  </Text>
                </Block>
              ) : null}
            </Block>
          </Block>
        );
    }
  }
}

Card.propTypes = {
  item: PropTypes.object,
  horizontal: PropTypes.bool,
  full: PropTypes.bool,
  ctaColor: PropTypes.string,
  imageStyle: PropTypes.any,
  ctaRight: PropTypes.bool,
  titleStyle: PropTypes.any,
  textBodyStyle: PropTypes.any,
};

const styles = StyleSheet.create({
  card: {
    backgroundColor: '#F2F2F3',
    marginVertical: theme.SIZES.BASE,
    borderWidth: 0,
    minHeight: 114,
    marginBottom: 4,
  },
  cardTitle: {
    paddingHorizontal: 9,
    paddingTop: 7,
    paddingBottom: 2,
    fontSize: 24,
    textAlign: 'center',
    fontWeight: 'bold',
  },
  cardTitle2: {
    paddingHorizontal: 9,
    paddingTop: 7,
    paddingBottom: 2,
    fontSize: 24,
    textAlign: 'center',
    fontWeight: 'bold',
    color: '#0f1e2e'
  },
  cardDescription: {
    // padding: theme.SIZES.BASE / 2
  },
  imageContainer: {
    borderRadius: 4,
    elevation: 1,
    overflow: 'hidden',
  },
  image: {
    // borderRadius: 3,
  },
  horizontalImage: {
    height: 200,
    width: 'auto',
  },
  horizontalStyles: {
    borderTopRightRadius: 0,
    borderBottomRightRadius: 0,
  },
  verticalStyles: {
    borderBottomRightRadius: 0,
    borderBottomLeftRadius: 0,
  },
  fullImage: {
    height: 215,
  },
  shadow: {
    shadowColor: '#8898AA',
    shadowOffset: { width: 0, height: 1 },
    shadowRadius: 6,
    shadowOpacity: 0.1,
    elevation: 2,
  },
  articleButton: {
    fontFamily: 'montserrat-bold',
    paddingHorizontal: 9,
    paddingVertical: 7,
  },
});

export default withNavigation(Card);
