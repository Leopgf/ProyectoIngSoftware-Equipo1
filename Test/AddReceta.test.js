import React from 'react';

import renderer from 'react-test-renderer';

import {
  StyleSheet,
  Dimensions,
  ScrollView,
  ImageBackground,
} from 'react-native';
import { Block, Text, theme, Button as GaButton } from 'galio-framework';
import { Images, nowTheme } from '../Vista/constants';
import LoadingView from 'react-native-loading-view';
import ModalDropdown from 'react-native-modal-dropdown';
import { Button, Icon, Input } from '../Vista/components';
import ImagePickerExample from '../Vista/components/ImagePicker';

const { width, height } = Dimensions.get('screen');


describe('<AddReceta />', () => {
  //
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <LoadingView
          loading={true}
          size="large"
          style={styles.cargar}
          text="Cargando Publicar Receta..."
        >
          <Block flex middle>
            <ImageBackground
              source={Images.AddReceta}
              style={styles.imageBackgroundContainer}
              imageStyle={styles.imageBackground}
            >
              <Block flex middle>
                <Block style={styles.registerContainer}>
                  <ScrollView
                    showsVerticalScrollIndicator={false}
                    contentInset={{ top: 30, left: 0, bottom: 60, right: 0 }}
                  >
                    <Block flex space="evenly">
                      <Block flex={0.4} middle>
                        <Block flex={0.5} middle>
                          <Text
                            style={{
                              fontFamily: 'montserrat-bold',
                              textAlign: 'center',
                              fontWeight: '500',
                              fontSize: 24,
                            }}
                            color="#0f1e2e"
                            size={24}
                          >
                            Publicar Receta
                          </Text>
                        </Block>
                      </Block>
                      <Block flex={0.1} middle style={{ marginTop: 10 }}>
                        <Text color="#0f1e2e" size={16}>
                          ¡Publica tus recetas con nosotros!
                        </Text>
                      </Block>
                      <Block flex={1} middle space="between">
                        <Block center flex={0.9}>
                          <Block flex space="between">
                            <Block>
                              <Block flex style={{ marginTop: 10 }}>
                                <ImagePickerExample />
                              </Block>
                              <Block flex center width={width * 0.8}>
                                <Input
                                  placeholder="Título de la Receta"
                                  style={styles.inputs}
                                  iconContent={<Icon size={18} name="tag" family="AntDesign" />}
                                />
                              </Block>
                              <Block flex center width={width * 0.8}>
                                <Input
                                  placeholder="Descripción de la Receta"
                                  style={styles.inputs}
                                  iconContent={<Icon size={18} name="text" family="Entypo" />}
                                />
                              </Block>
                              <Block flex center width={width * 0.8}>
                                <Input
                                  placeholder="Porción por defecto de la receta"
                                  keyboardType="numeric"
                                  style={styles.inputs}
                                  iconContent={<Icon size={18} name="text" family="Entypo" />}
                                />
                              </Block>
                              <Block flex center width={width * 0.8}>
                                <Input
                                  placeholder="Unidad de las porciones de la Receta"
                                  style={styles.inputs}
                                  iconContent={<Icon size={18} name="text" family="Entypo" />}
                                />
                              </Block>
                              <Block flex center width={width * 0.8}>
                                <Block row flex center>
                                  <Text color="#0f1e2e" size={16} style={{ marginTop: 10 }}>
                                    Categoría:
                                  </Text>
                                  <ModalDropdown
                                    ref="dropdown"
                                    defaultValue={'Seleccione'}
                                    textStyle={styles.dropdownText}
                                    style={styles.dropdown}
                                    dropdownStyle={styles.dropdownOption}
                                    options={[]}
                                  />
                                </Block>
                                <Block flex center width={width * 0.8}>
                                  <Block
                                    row
                                    flex
                                    center
                                    style={{
                                      marginTop: 15,
                                      backgroundColor: '#ffffff',
                                      borderRadius: 50,
                                      alignSelf: 'center',
                                      padding: 10,
                                    }}
                                  >
                                    <Text color="#e63746" size={12}>
                                      Click en + para añadir un paso y en - para eliminar un paso
                                      específico.
                                    </Text>
                                  </Block>
                                  <Block row flex center>
                                    <Text color="#0f1e2e" size={16} style={{ marginTop: 10 }}>
                                      Pasos de la receta:
                                    </Text>
                                    <Button
                                      style={{
                                        fontFamily: 'montserrat-bold',
                                        borderRadius: nowTheme.SIZES.BASE * 1.5,
                                        width: 30,
                                        height: 30,
                                        marginTop: 10,
                                        marginLeft: 5,
                                      }}
                                      color="primary"
                                      round
                                    >
                                      +
                                    </Button>
                                  </Block>
                                
                                    <Block flex syle={{ flexDirection: 'row' }} >
                                      <Block row flex center style={{ marginTop: 10 }} >
                                        <Input
                                          placeholder={`Paso`}
                                          style={styles.inputs}
                                          iconContent={
                                            <Icon size={18} name="tag" family="AntDesign" />
                                          }
                                          value={''}
                                          
                                        />
                                        <Button
                                          style={{
                                            fontFamily: 'montserrat-bold',
                                            borderRadius: nowTheme.SIZES.BASE * 1.5,
                                            width: 30,
                                            height: 30,
                                            marginTop: 0,
                                            marginLeft: 5,
                                          }}
                                          color="primary"
                                          round
                                        >
                                          -
                                        </Button>
                                      </Block>
                                    </Block>
                                
                                </Block>
                                <Block flex center width={width * 0.8} style={{ marginTop: 20 }}>
                                  <Block
                                    row
                                    flex
                                    center
                                    style={{
                                      marginTop: 15,
                                      margin: 5,
                                      backgroundColor: '#ffffff',
                                      borderRadius: 50,
                                      alignSelf: 'center',
                                      padding: 10,
                                    }}
                                  >
                                    <Text color="#e63746" size={12}>
                                      Click en + para añadir un ingrediente y en - para eliminar un
                                      ingrediente específico.
                                    </Text>
                                  </Block>
                                  <Block row flex center>
                                    <Text color="#0f1e2e" size={16}>
                                      Ingredientes:
                                    </Text>
                                    <Button
                                      style={{
                                        fontFamily: 'montserrat-bold',
                                        borderRadius: nowTheme.SIZES.BASE * 1.5,
                                        width: 30,
                                        height: 30,
                                        marginLeft: 10,
                                      }}
                                      color="primary"
                                      round
                                    >
                                      +
                                    </Button>
                                  </Block>

                                  
                                    <Block flex syle={{ flexDirection: 'row' }} >
                                      <Block flex center row>
                                        <ModalDropdown
                                          ref="dropdown"
                                          defaultValue={'Cantidad'}
                                          textStyle={styles.dropdownText}
                                          style={styles.dropdown}
                                          dropdownStyle={styles.dropdownOption}
                                          options={['Al gusto', 'Cantidad']}
                                         
                                        />
                                        <Button
                                          style={{
                                            fontFamily: 'montserrat-bold',
                                            borderRadius: nowTheme.SIZES.BASE * 1.5,
                                            width: 30,
                                            height: 30,
                                            marginLeft: 2,
                                            marginTop: 10,
                                          }}
                                          color="primary"
                                          round
                                          
                                        >
                                          -
                                        </Button>
                                      </Block>
                                     
                                        <Input
                                          placeholder={`Cantidad del ingrediente`}
                                          style={styles.inputs}
                                          keyboardType="numeric"
                                          iconContent={
                                            <Icon size={18} name="tag" family="AntDesign" />
                                          }
                                          value={''}
                                          
                                        />
                                      <Input
                                        placeholder={`Ingrediente `}
                                        style={styles.inputs}
                                        iconContent={
                                          <Icon size={18} name="tag" family="AntDesign" />
                                        }
                                        value={''}
                                        
                                      />
                                    </Block>
                                </Block>
                              </Block>
                            </Block>
                            <Block center style={{ marginTop: 10 }}>
                              <Button
                                style={{
                                  fontFamily: 'montserrat-bold',
                                  borderRadius: nowTheme.SIZES.BASE * 1.5,
                                  width: 200,
                                }}
                                color="primary"
                                round
                                
                              >
                                <Text
                                  style={{ fontFamily: 'montserrat-bold' }}
                                  size={14}
                                  color={nowTheme.COLORS.WHITE}
                                >
                                  PUBLICAR
                                </Text>
                              </Button>
                            </Block>
                          </Block>
                        </Block>
                      </Block>
                    </Block>
                  </ScrollView>
                </Block>
              </Block>
            </ImageBackground>
          </Block>
        </LoadingView>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});


const styles = StyleSheet.create({
    imageBackgroundContainer: {
      width: width,
      height: height,
      padding: 0,
      zIndex: 1,
    },
    imageBackground: {
      width: width,
      height: height,
    },
    dropdown: {
      marginLeft: 15,
      alignSelf: 'center',
      width: width / 3,
      height: height / 24,
      marginTop: 10,
      alignItems: 'center',
      right: 8,
      borderWidth: 0,
      borderRadius: 15,
      backgroundColor: '#e63746',
    },
  
    dropdownOption: {
      width: width / 5,
      alignItems: 'center',
      borderColor: '#e63746',
      borderRadius: 10,
      borderWidth: 2,
      borderRadius: 3,
    },
    dropdownText: {
      marginVertical: 7,
      marginHorizontal: 6,
      fontSize: 15,
      color: '#c2c1c1',
      textAlign: 'center',
      textAlignVertical: 'center',
    },
    registerContainer: {
      marginTop: 55,
      width: width * 0.9,
      height: height < 812 ? height * 0.8 : height * 0.8,
      backgroundColor: nowTheme.COLORS.WHITE,
      borderRadius: 4,
      shadowColor: nowTheme.COLORS.BLACK,
      shadowOffset: {
        width: 0,
        height: 4,
      },
      shadowRadius: 8,
      shadowOpacity: 0.1,
      elevation: 1,
      overflow: 'hidden',
    },
    socialConnect: {
      backgroundColor: nowTheme.COLORS.WHITE,
      // borderBottomWidth: StyleSheet.hairlineWidth,
      // borderColor: "rgba(136, 152, 170, 0.3)"
    },
    socialButtons: {
      width: 120,
      height: 40,
      backgroundColor: '#fff',
      shadowColor: nowTheme.COLORS.BLACK,
      shadowOffset: {
        width: 0,
        height: 4,
      },
      shadowRadius: 8,
      shadowOpacity: 0.1,
      elevation: 1,
    },
    socialTextButtons: {
      color: nowTheme.COLORS.PRIMARY,
      fontWeight: '800',
      fontSize: 14,
    },
    inputIcons: {
      marginRight: 12,
      color: nowTheme.COLORS.ICON_INPUT,
    },
    inputs: {
      width: width * 0.65,
      borderWidth: 1,
      borderColor: '#E3E3E3',
      borderRadius: 21.5,
      alignSelf: 'center',
    },
    passwordCheck: {
      paddingLeft: 2,
      paddingTop: 6,
      paddingBottom: 15,
    },
    createButton: {
      width: width * 0.5,
      marginTop: 25,
      marginBottom: 40,
    },
    social: {
      width: theme.SIZES.BASE * 3.5,
      height: theme.SIZES.BASE * 3.5,
      borderRadius: theme.SIZES.BASE * 1.75,
      justifyContent: 'center',
      marginHorizontal: 10,
    },
  });