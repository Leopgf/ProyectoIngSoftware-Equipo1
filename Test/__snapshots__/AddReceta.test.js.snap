// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddReceta /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": 10,
      },
      undefined,
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
    style={
      Array [
        undefined,
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "fontSize": 18,
          "fontWeight": "600",
          "marginTop": 10,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Cargando Publicar Receta...
  </Text>
</View>
`;
